### PR TITLE
Add guidance on PRs and git commits

### PR DESCRIPTION
**Pull request by cjellick**
We need to make the expectations around git commits and the process for
pull requests more clear to contributors.

### CODE REVIEW

**Comment by brandond** [comment]
LGTM except for no PR template!

**Review by brandond — APPROVED** [review]

**Review by galal-hussein — APPROVED** [review]
LGTM

**Comment by ibuildthecloud** [comment]
I personally suck at commit messages, but from the reference referenced post this recommendation is really good

> A properly formed Git commit subject line should always be able to complete the following sentence:
>
>    If applied, this commit will your "subject line here"

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,6 +10,59 @@ If you're interested in contributing documentation, please note the following:
...
 - Pull requests are submitted to the K3s documentation source in the [Rancher docs repository.](https://github.com/rancher/docs/) The K3s docs content is in the `content/k3s/` directory.
 
+## Pull request and git commit guidance
+
+### Opening PRs and organizing commits
+PRs should generally address only 1 issue at a time. If you need to fix two bugs, open two separate PRs. This will keep the scope of your pull requests smaller and allow them to be reviewed and merged more quickly.
+
+When possible, fill out as much detail in the pull request template as is reasonable. Most important is to reference the GitHub issue that you are addressing with the PR.
+
+**NOTE:** GitHub has [a feature](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that will automatically close issues referenced with a keyword (such as "Fixes") by a PR or commit once the PR/commit is merged. Don't use these keywords. We don't want issues to be automatically closed. We want our testers to independently verify and close them.
```
Comment (dramich) [inline CONTRIBUTING.md:20]: Great note. Curious if we can turn this off somehow or have a bot fix this for us by editing the PR

**Review by dramich — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (ibuildthecloud) [reply CONTRIBUTING.md:20]: Nobody else does this.... Which makes me think we are doing something wrong.

**Review by briandowns — APPROVED** [review]
LGTM

File: CONTRIBUTING.md (hunk 1 above)
Comment (cjellick) [reply CONTRIBUTING.md:20]: It cannot be turned off in github (yet).

darren, what part do you think we're doing wrong? trying to keep issues open?

Comment (ibuildthecloud) [reply CONTRIBUTING.md:20]: Yeah, keeping issues open.  But it's going to go into a much, much bigger discussion. Probably shouldn't have said anything as this PR isn't the place to have that tangent.

Comment (cjellick) [reply CONTRIBUTING.md:20]: yep, no problem. i knew this would generate "fix the process" comments.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -26,6 +26,12 @@ The other exception to this single-commit rule is if your PR includes a change t
 ### Writing good commit messages
 Git commit messages should explain the how and why of your change and be separated into a brief subject line followed by a more detailed body. When in doubt, follow this guide for good commit messages and you can’t go wrong: https://chris.beams.io/posts/git-commit/.
 
+One particularly useful point made in the above guide is regarding commit subject lines:
+
+A properly formed Git commit subject line should always be able to complete the following sentence:
+
+- If applied, this commit will <ins>your subject line here</ins>
+
 One simple but effective pattern to follow for commits is the “problem / solution” pattern. It looks like this:
```
Comment (brandond) [inline CONTRIBUTING.md:35]: Starting two paragraphs in a row with 'One', and saying 'pattern' twice in a sentence, sounds a little off to me... how about:

```suggestion
A simple but effective convention to follow for commits is the “problem / solution” pattern. It looks like this:
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -10,6 +10,65 @@ If you're interested in contributing documentation, please note the following:
...
+The other exception to this single-commit rule is if your PR includes a change to a vendored dependency or generated code. To make reviewing easier, these changes should be segregated into their own commit. Note that as we migrate from using the vendor directory to a pure go module model for our projects, this will be less of an issue.
+
+### Writing good commit messages
+Git commit messages should explain the how and why of your change and be separated into a brief subject line followed by a more detailed body. When in doubt, follow this guide for good commit messages and you can’t go wrong: https://chris.beams.io/posts/git-commit/.
+
+One particularly useful point made in the above guide is regarding commit subject lines:
+
+A properly formed Git commit subject line should always be able to complete the following sentence:
+
+- If applied, this commit will <ins>your subject line here</ins>
```
Comment (brandond) [inline CONTRIBUTING.md:33]:
```suggestion
> A properly formed Git commit subject line should always be able to complete the following sentence:
> 
> - If applied, this commit will <ins>your subject line here</ins>
```

**Review by brandond — CHANGES_REQUESTED** [review]

**Pull request merged**